src/current_input.rs

```diff
@@ -140,6 +143,15 @@ impl CurrentInput {
             _ => {}
         }
     }
+
+    pub fn handle_device_event(&mut self, event: &DeviceEvent) {
+        match event {
+            DeviceEvent::MouseMotion { delta, .. } => {
+                self.mouse_diff = Some((delta.0 as f32, delta.1 as f32))
```

Comment:
I think that this implementation will give inaccurate results when we receive multiple MouseMotion events within a single step.
To solve this we should detect when mouse_diff is already Some and then add the new delta's on top of the old deltas.